gateway/src/apicast/policy/rate_limit/rate_limit.lua

```diff
@@ -0,0 +1,205 @@
+local policy = require('apicast.policy')
+local _M = policy.new('Rate Limit Policy')
+
+local resty_limit_conn = require('resty.limit.conn')
+local resty_limit_req = require('resty.limit.req')
+local resty_limit_count = require('resty.limit.count')
+
+local ngx_semaphore = require "ngx.semaphore"
+local limit_traffic = require "resty.limit.traffic"
+local ts = require ('apicast.threescale_utils')
+local tonumber = tonumber
+local next = next
+local shdict_key = 'limiter'
+
+local new = _M.new
+
+local traffic_limiters = {
+  connections = function(config)
+    return resty_limit_conn.new(shdict_key, config.conn, config.burst, config.delay)
+  end,
+  leaky_bucket = function(config)
+    return resty_limit_req.new(shdict_key, config.rate, config.burst)
+  end,
+  fixed_window = function(config)
+    return resty_limit_count.new(shdict_key, config.count, config.window)
+  end
+}
+
+local function try(f, catch_f)
+  local status, exception = pcall(f)
+  if not status then
+    catch_f(exception)
+  end
+end
+
+local function init_limiter(config)
+  local lim, limerr
+  try(
```

Comment:
This is not really idiomatic Lua and we really should not be expecting exceptions.

All resty code will return an error as a second return value, so we can just use that.
Otherwise our code should not crash because that would trash performance of the whole gateway.